diff --git a/target/linux/ar71xx/image/generic.mk b/target/linux/ar71xx/image/generic.mk
index f08f1e8..08eb99f 100644
--- a/target/linux/ar71xx/image/generic.mk
+++ b/target/linux/ar71xx/image/generic.mk
@@ -889,7 +889,7 @@ endef
 TARGET_DEVICES += mc-mac1200r
 
 define Device/mc-mw4530r
-$(Device/tplink-8mlzma)
+$(Device/tplink-16mlzma)
   DEVICE_TITLE := Mercury MW4530R
   DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
   BOARDNAME := MC-MW4530R
